samples/nrf9160/sms/src/main.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdio.h>
+#include <kernel.h>
+#include <string.h>
+#include <modem/sms.h>
+
+
+static void sms_callback(struct sms_data *const data, void *context)
+{
+	if (data == NULL) {
+		printk("sms_callback with NULL data\n");
+	}
+
+	if (data->type == SMS_TYPE_DELIVER) {
+		/* When SMS message is received, print information */
+		struct sms_deliver_header *header = &data->header.deliver;
+		printk("\nSMS received:\n");
+		printk("\tTime:   %02d-%02d-%02d %02d:%02d:%02d\n",
+			header->time.year,
+			header->time.month,
+			header->time.day,
+			header->time.hour,
+			header->time.minute,
+			header->time.second);
+
+		printk("\tText:   '%s'\n", data->data);
+		printk("\tLength: %d\n", data->data_len);
+
+		if (header->app_port.present) {
+			printk("\tApplication port addressing scheme: dest_port=%d, src_port=%d\n",
+				header->app_port.dest_port,
+				header->app_port.src_port);
+		}
+		if (header->concatenated.present) {
+			printk("\tConcatenated short message: ref_number=%d, msg %d/%d\n",
+				header->concatenated.ref_number,
+				header->concatenated.seq_number,
+				header->concatenated.total_msgs);
+		}
+	} else if (data->type == SMS_TYPE_STATUS_REPORT) {
+		printk("SMS status report received\n");
+		return;
+	} else {
+		printk("SMS protocol message with unknown type received\n");
+	}
+}
+
+void main(void)
+{
+	printk("\nSMS sample starting\n");
+
+	int handle = sms_register_listener(sms_callback, NULL);
+	if (handle) {
+		printk("sms_register_listener returned err: %d\n", handle);
+		return;
+	}
+
+	printk("SMS sample is ready for receiving messages\n");
+
+	/* SMS sending is commented out here as destination phone number that
+	 * we should use is unknown. User can tweak the code, e.g., to send
+	 * the message to his/her personal phone.
+	 */
+	printk("\nIf you want to send an SMS, please find this line from the code.\n"
+		"Then, uncomment next line and change your phone number in there.\n");
+	/*
+	int ret = sms_send("000000000000", "SMS sample: testing"); 
```

Comment:
I'm conflicted about this sample.
On one hand I really like that it is this simple and hard coded. Easy to understand, to modify and copy paste.
But on the other hand, it can not work out of the box in this case.
Perhaps a small shell is needed here similar to how you can enter number and message in modem shell application.
Anyone else have thoughts about this? Maybe fetch the number from proj.cfg?

Reply:
Kconfig variable for number

Reply:
This has been changed. Please check and if it's fine, resolve this conversation.

Reply:
Tested. looks good to me. 👍